src/platform/nodejs/lib/util/crypto.ts

```diff
@@ -198,7 +198,10 @@ var CryptoFactory = function (bufferUtils: typeof BufferUtils) {
 
       generateRandom((keyLength || DEFAULT_KEYLENGTH) / 8, function (err, buf) {
         if (callback !== undefined) {
-          callback(err ? ErrorInfo.fromValues(err) : null, buf);
+          const errorInfo = err
+            ? new ErrorInfo('Failed to generate random key: ' + err.message, 500, 50000, err)
```

Comment:
I had absolutely no idea what would be an appropriate code and status code to use here — any thoughts @owenpearson?

Reply:
You can give status code used for client error

Reply:
HttpStatusCode.BadRequest

Reply:
Hey @sacOO7, thanks for that, I've changed the `code` to 50000 to match what you linked to. I'm a bit confused about why "bad request" is the right thing to use, though, but if that's what we're doing then I guess I won't question it  🤷‍♂️